### ISSUE TITLE
[bug] Latentscope 0.3.1 web server did not load

### ISSUE THREAD

**Issue by hydrosquall** [enjalot/latent-scope#48]
In a fresh virtual environment using Python 3.12.1 on Mac OS, this loads content at http://localhost:5001/

```
pip install latentscope===0.3.0
ls-serve
```

However this returns an error message at localhost:5001

```
pip install latentscope===0.3.1
ls-serve
```

<img width="733" alt="image" src="https://github.com/user-attachments/assets/b862a584-278e-4de2-8176-927c7779f469">

I looked at the [diff](https://github.com/enjalot/latent-scope/compare/v0.3.0...v0.3.1) and couldn't figure out what changed between versions, I'm also not getting much info out of the console. I'll stay pinned to 0.3.0 for now, but update this thread if I find out what happened.

**Comment by enjalot** [comment]
that's really strange. so fresh install in a venv and you get that? I've had an issue where I see that error message if I refresh on a deep URL, but not starting from the home page at localhost:5001

I don't see anything that should have affected that in the diff. I'll try to reproduce with a clean venv too.

**Comment by hydrosquall** [comment]
I looked into it deeper, somehow my package 0.3.1 was being installed from my local github copy rather than directly from `pypi`. I'm able to run the site if I visit `http://localhost:5174/` while running the `ls-serve` API at the same time. Thanks for having a look, we can probably close this issue.

>  I've had an issue where I see that error message if I refresh on a deep URL

I noticed this as well specifically when on the "explore" pages, I think it's a different issue than the one I bumped into.

I think this doesn't happen on the vite server because everything gets managed by the browser router, but in the flask app this affects all the `/datasets/<DATASET_NAME>/explore/<SCOPE_NAME>` routes. 

I think this [catchall block](https://github.com/enjalot/latent-scope/blob/b07a27bc6685f3b4b4287251bf094ee8e1767dc2/latentscope/server/app.py#L212-L213) might want to redirect to `index.html` rather than directing to alternate dist files if what's being served looked like a React page rather than an asset like an img/js/css file.
